### PR TITLE
Store tags and description embedding in vector database

### PR DESCRIPTION
**Pull request by V8-Luky**

### CODE REVIEW

File: backend/integration.py [hunk 1]
```diff
@@ -67,4 +84,10 @@ def _compare_games(self, base_metadata: dict, comparable_metadata: dict) -> Game
         guessed_game = Game.from_metadata(comparable_metadata)
 
         return GameComparison(target_game, guessed_game)
+    
+    def _get_weighted_similarity(self, similarity_scores: list[float], weights: list[float]) -> float:
+        return sum([score * weight for score, weight in zip(similarity_scores, weights)])
```
Comment (SilvanHSLU) [inline backend/integration.py:89]: That's such a beautiful pythonic way of implementing that <3

File: embedding/tags_embedder.py [hunk 2]
```diff
@@ -12,9 +12,9 @@ def __init__(self):
     def create_embeddings(self, games: DataFrame) -> ndarray:
         game_tags = games["Tags"].apply(self._tokenize)
         embeddings = game_tags.apply(lambda x: np.mean([self._model[word] for word in x if word in self._model], axis=0)).to_numpy()
-        for embedding in embeddings:
-            if embedding.shape[0] != 300:
-                embedding = np.zeros(300)
+        for i, embedding in enumerate(embeddings):
+            if not embedding.shape or embedding.shape[0] != 300:
+                embeddings[i] = np.zeros(300)
```
Comment (SilvanHSLU) [inline embedding/tags_embedder.py:17]: Thank you ;)

**Review by SilvanHSLU — APPROVED** [review]

**Comment by SilvanHSLU** [comment]
@V8-Luky Ah shit didn't know it's blocking the merge haha just wanted to appreciate the work :D

**Comment by V8-Luky** [comment]
> @V8-Luky Ah shit didn't know it's blocking the merge haha just wanted to appreciate the work :D

No worries, thanks for the review :)

**Pull request merged**